pkg/controller/build/imagebuilder/base.go

```diff
@@ -0,0 +1,342 @@
+func newBaseImageBuilderWithCleaner(kubeclient clientset.Interface, mcfgclient mcfgclientset.Interface, mosb *mcfgv1alpha1.MachineOSBuild, mosc *mcfgv1alpha1.MachineOSConfig, builder buildrequest.Builder) (*baseImageBuilder, Cleaner) {
+	b := newBaseImageBuilder(kubeclient, mcfgclient, mosb, mosc, builder)
+	return b, &cleanerImpl{
+		baseImageBuilder: b,
+	}
+}
+
+// Represents the successful conditions for a MachineOSBuild.
+func (b *baseImageBuilder) succeededConditions() []metav1.Condition {
+	return []metav1.Condition{
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildPrepared),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Prepared",
+			Message: "Build Prepared and Pending",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuilding),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Building",
+			Message: "Image Build In Progress",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildFailed),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Failed",
+			Message: "Build Failed",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildInterrupted),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Interrupted",
+			Message: "Build Interrupted",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildSucceeded),
+			Status:  metav1.ConditionTrue,
+			Reason:  "Ready",
+			Message: "Build Ready",
+		},
+	}
+}
+
+// Represents the pending conditions for a MachineOSBuild.
+func (b *baseImageBuilder) pendingConditions() []metav1.Condition {
+	return []metav1.Condition{
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildPrepared),
+			Status:  metav1.ConditionTrue,
+			Reason:  "Prepared",
+			Message: "Build Prepared and Pending",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuilding),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Building",
+			Message: "Image Build In Progress",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildFailed),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Failed",
+			Message: "Build Failed",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildInterrupted),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Interrupted",
+			Message: "Build Interrupted",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildSucceeded),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Ready",
+			Message: "Build Ready",
+		},
+	}
+}
+
+// Represents the running conditions for a MachineOSBuild.
+func (b *baseImageBuilder) runningConditions() []metav1.Condition {
+	return []metav1.Condition{
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildPrepared),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Prepared",
+			Message: "Build Prepared and Pending",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuilding),
+			Status:  metav1.ConditionTrue,
+			Reason:  "Building",
+			Message: "Image Build In Progress",
+		},
+		{
+			Type:    string(mcfgv1alpha1.MachineOSBuildFailed),
+			Status:  metav1.ConditionFalse,
+			Reason:  "Failed",
```

Comment:
non-blocking thought:
Similar to the Node Controller, we could `HandleErr()` which would append the reason annotation with the Truncated Error resulting from the pod/job and enqueue the object in the queue.

Reply:
That would be a useful addition, for sure. I'd like to consider that once we get the Jobs work landed because there might be some additional status reporting which could be useful.